packages/tenant-process/src/services/tenantService.ts

```diff
@@ -529,15 +529,28 @@ export function tenantServiceBuilder(
           correlationId
         );
 
-        await repository.createEvents([
+        const createdEvents = await repository.createEvents([
           tenantCertifiedAttributeAssignedEvent,
           tenantKindUpdatedEvent,
         ]);
-      } else {
-        await repository.createEvent(tenantCertifiedAttributeAssignedEvent);
-      }
 
-      return updatedTenant;
+        const newVersion = createdEvents.reduce(
+          (acc, event) => Math.max(acc, event.newVersion),
+          0
+        );
```

Comment:
More readable using map IMO

```suggestion
const newVersion = Math.max(0, ...createdEvents.map(event => event.newVersion));
```